addons/binding/org.openhab.binding.kodi/ESH-INF/thing/thing-types.xml

```diff
@@ -168,4 +175,16 @@
 		<description>Media type of the current file</description>
 		<state readOnly="true" pattern="%s"></state>
 	</channel-type>
+	<channel-type id="thumbnail" advanced="true">
+		<item-type>String</item-type>
```

Comment:
Err, seeing this feature I would have very much expected that this supports the Image item type and not String.

You might be interested in https://github.com/eclipse/smarthome/issues/3448 - once it is available, this should make it pretty easy to adapt your implementation here. Would you agree?